### ISSUE TITLE
NeedBraces: False Positives for LITERAL_CASE and LITERAL_DEFAULT

### ISSUE THREAD

**Issue by gkeighren** [checkstyle/checkstyle#4764]
https://checkstyle.org/config_blocks.html#NeedBraces

```
/var/tmp $ javac NeedBracesSwitchTest.java

```

```
/var/tmp $ cat NeedBracesSwitchTest.java

public class NeedBracesSwitchTest {

  // NeedBraces should not raise any violations for this code
  // Violation raised by Checkstyle 6.4+

  // Specific cases tested:
  // Raises violation: 8.0, 7.8.2, 7.7, 7.6, 7.5.1, 6.19, 6.7, 6.4.1, 6.4
  // No violations: 5.9, 6.1, 6.2, 6.3

  public String aMethod(int val) {
      switch (val) {
              case 0: {
                      return "zero";
                  }
              case 1: {
                      return "one";
                  }
              default: {
                      return "non-binary";
                  }
          }
  }

}
```

```
/var/tmp $ cat config.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="NeedBraces">
            <property name="tokens" value="LITERAL_CASE, LITERAL_DEFAULT"/>
        </module>
    </module>
</module>
```

```
/var/tmp $ java -jar checkstyle-8.0-all.jar -c config.xml NeedBracesSwitchTest.java

Starting audit...
[ERROR] C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:12: 'case' construct must use '{}'s. [NeedBraces]
[ERROR] C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:15: 'case' construct must use '{}'s. [NeedBraces]
[ERROR] C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:18: 'default' construct must use '{}'s. [NeedBraces]
Audit done.
Checkstyle ends with 3 errors.
```

```
/var/tmp $ java -jar checkstyle-6.4-all.jar -c config.xml NeedBracesSwitchTest.java
Starting audit...
C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:12: 'case' construct must use '{}'s.
C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:15: 'case' construct must use '{}'s.
C:\Users\gavin.keighren\Desktop\NeedBracesSwitchTest.java:18: 'default' construct must use '{}'s.
Audit done.
```

```
/var/tmp $ java -jar checkstyle-6.3-all.jar -c config.xml NeedBracesSwitchTest.java
Starting audit...
Audit done.
```

---------------

I expect no violations reported. Based on local testing, the incorrect violations started being reported with Checkstyle 6.4. Please see comment in java test file for list of locally tested versions and the results.

**Comment by rnveach** [comment]
> the incorrect violations started being reported with Checkstyle 6.4

Though we don't specify when acceptable tokens were added in config `case` and `default` were not added as acceptable tokens until https://github.com/checkstyle/checkstyle/commit/abe2b200312ee72aa552026b1a34e8bd72bfdc4e#diff-95edbb4da33f2c258c9d24deab6b19ec .
Before this commit, acceptable tokens was same as `default` which didn't list `case` and `default`.

I'm not sure why there is no error specifying this when doing regression with 6.3, but it must be a bug of some kind in old code as there will be error in 6.4 if it was not part of acceptable.

**Comment by romani** [comment]
Fix is merged
